tls/s2n_handshake.c

```diff
@@ -155,3 +155,18 @@ int s2n_conn_update_required_handshake_hashes(struct s2n_connection *conn)
 
     return 0;
 }
+
+int s2n_conn_find_name_matching_certs(struct s2n_connection *conn)
+{
+    const char *name = conn->server_name;
+    for (int i = 0; i < conn->config->num_certificates; i++) {
```

Comment:
This is a naive implementation that's fine for a low max certificate limit. I'm keen on optimizing this lookup in a future, standalone change.